src/Runner.Listener/JobDispatcher.cs

```diff
@@ -661,6 +661,7 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
                 try
                 {
                     request = await runnerServer.RenewAgentRequestAsync(poolId, requestId, lockToken, orchestrationId, token);
+                    UpdateAgentNameIfNeeded(request.ReservedAgent?.Name);
```

Comment:
can we also move this to line 670 so we will only check the `.runner` file once per-job.